source/develop/release-management/features/storage/4k-blocks.md

```diff
@@ -0,0 +1,93 @@
+---
+title: 4k block device support for the storage domains
+category: feature
+authors: dchaplyg
+feature_name: 4k block device support for the storage domains
+feature_modules: vdsm
+feature_status: Planning
+---
+
+# 4k block device support for the storage domains
+
+# Summary
+
+VDSM assumes, that 512b is a minimally accessible block on the storage device, which can be written or read atomically. That was true for a long time, but now most of the storage devices changed their block size to the 4096 bytes (4 kilobytes). It is still compatible with VDSM, case while we are requesting 512 atomic read or write, underlying storage will atomically read 4096 bytes, throw away 3584 bytes and return just 512. As it still acts same way, it has no impact on the VDSM. 
+
+Unfortunately, some 4k block storage devices either refuse to read/write smaller blocks atomically (some SAN appliances) or have a performance impact while emulating 512b block device (like VDO). 
```

Comment:
We need more info on this issue. How widespread are drives or storage systems that do not support 512 bytes emulation?

Reply:
VDO:)

Will try to find more brands/specific systems details